### PR TITLE
Api/service user search

### PR DESCRIPTION
**Pull request by mugnivenko**

### CODE REVIEW

File: api/SWR/CarePackage/useCarePackageApi.js [hunk 1]
```diff
@@ -28,7 +28,43 @@ const useCarePackageApi = {
...
+    firstName,
+    lastName,
+    hackneyId,
+    dateOfBirth,
+    postcode,
+    pageNumber,
+    pageSize,
+    orderBy
+  }, shouldFetch) => (
+    useGetData(shouldFetch ? `/service-user/search${getQueryParamsFromObject({
```
Comment (vadim-kirpun) [inline api/SWR/CarePackage/useCarePackageApi.js:48]: About query params for that and other requests
`useGetData` was supposed to be used for simple requests without any params. And although the current approach is working with this query builder - this is not how it is supposed to be done from the SWR perspective.

https://swr.vercel.app/docs/arguments#multiple-arguments
Here is an example of how it should be done for multiple arguments. I think we should consider refactoring it in future
cc @akeamwoodsND

File: api/SWR/package/usePrimarySupportReason.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
 import useGetData from '../useGetData';
 
-const usePrimarySupportReason = () => useGetData('/primary-support-reasons');
+const usePrimarySupportReason = () => {
+  const response = useGetData('/primary-support-reasons');
+
+  return { ...response, primarySupportReasonLoading: !!response.data };
```
Comment (vadim-kirpun) [inline api/SWR/package/usePrimarySupportReason.js:6]: There is a `isLoading` prop provided by `useGetData`

File: api/SWR/useGetData.js [hunk 3]
```diff
@@ -6,12 +6,11 @@ const useGetData = (url, errorMessage, initialData = []) => {
   const response = useSWR(url, {
     shouldRetryOnError: false,
     revalidateOnFocus: false,
-    initialData,
```
Comment (vadim-kirpun) [inline api/SWR/useGetData.js:9]: `initialData` is used in many requests to set `{}` as default

**Comment by akeamwoodsND** [comment]
Why is this PR so big? Which tasks does this cover?

**Review by vadim-kirpun — APPROVED** [review]

**Pull request merged**